### PR TITLE
chore: address issues in code review

### PR DESCRIPTION
**Pull request by SeisSerenata**
## Description
This PR includes several improvements to the batch processing functionality and error handling:

- Updates the batch processing endpoint URL to use the production domain
- Adds proper initialization for batch URL in AnyParser class
- Improves file validation in batch processing
- Enhances response models for batch operations
- Updates documentation regarding batch processing limitations

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement

## How Has This Been Tested?
- Tested file upload functionality with both existing

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Additional Notes

___

### **PR Type**
Bug fix, Documentation, Enhancement

___

### **Description**
- Updated `PUBLIC_BATCH_BASE_URL` to use the production domain for batch processing.
- Enhanced the `AnyParser` class by adding a `batch_url` parameter for better flexibility.
- Improved file upload handling in `BatchParser` by adding a validation check for file existence.
- Removed unnecessary debug print statements for cleaner code.
- Added docstrings to response models in `batch_parser.py` for better clarity and maintainability.
- Updated the README to include important notes about batch processing permissions and limitations.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>any_parser.py</strong><dd><code>Update batch URL and enhance AnyParser initialization.</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

any_parser/any_parser.py

<li>Updated <code>PUBLIC_BATCH_BASE_URL</code> to use the production domain.<br> <li> Added <code>batch_url</code> parameter to <code>AnyParser</code> class initialization.<br> <li> Updated <code>BatchParser</code> initialization to use the <code>batch_url</code> parameter.<br>

</details>

  </td>
  <td><a href="https://github.com/CambioML/any-parser/pull/72/files#diff-b580101abf5c2c14a05ba9e6a763e462c65136ba8e884488ddcd93561600346d">+8/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>batch_parser.py</strong><dd><code>Improve batch file handling and response model documentation.</code></dd></summary>
<hr>

any_parser/batch_parser.py

<li>Added validation to check if the file path exists before upload.<br> <li> Removed unnecessary print statements for cleaner code.<br> <li> Enhanced response models with docstrings for better clarity.<br>

</details>

  </td>
  <td><a href="https://github.com/CambioML/any-parser/pull/72/files#diff-e941c704ac7b7e5232d6b2e8e0c6375bee577c4a0c594948dbac6925932cb1cd">+16/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update README with batch processing permissions note.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

<li>Added a note about batch processing permissions for API keys.<br> <li> Improved documentation regarding batch processing limitations.<br>

</details>

  </td>
  <td><a href="https://github.com/CambioML/any-parser/pull/72/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +77,8 @@ markdown = ap.batches.retrieve(request_id)
 ```
 
 > ⚠️ **Note:** Batch extraction is currently in beta testing. Processing time may take up to 12 hours to complete.
+> 
+> ⚠️ **Important:** API keys generated from cambioml.com do not automatically have batch processing permissions. Please contact info@cambioml.com to request batch processing access for your API key.
```
Comment (lingjiekong) [inline README.md:81]: Nice!

**Review by lingjiekong — APPROVED** [review]
LGTM

**Comment by boqiny** [comment]
reopen to trigger workflow

**Review by lingjiekong — APPROVED** [review]
LGTM

**Pull request merged**